main.rb

```diff
@@ -1,21 +1,12 @@
 require_relative './app'
+require_relative './options_module'
 
 class Main
   def initialize
     @app = App.new
   end
 
-  def display_list
-    puts "
-        Please choose an option by entering a number:
-        1- List all books.
-        2- List all people.
-        3- Create a person.
-        4- Create a book.
-        5- Create a rental.
-        6- List all rentals for a given person id.
-        7- Exit"
-  end
+  include Options
 
```

Comment:
- [ ] All requests for input and input validation should've been placed in the **_main.rb_** file and/or classes/methods whose responsibilities are to request or validate input. Essentially, you decide to refactor your code such that you have a separate class or method for receiving input and input validation. [Remember this lesson on SOLID principles.](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

#### **Your code**
```ruby
 def create_person
    puts 'Do you want to create a student (1) or a teacher (2)? [Input a number]:'
    person_type = gets.chomp.to_i

    case person_type
    when 1
      create_student
    when 2
      create_teacher
    else
      invalid_input
    end
  end
```

Possible refactor

```ruby
# This method will be used across your project for getting user inputs. This will improve code reusablilty.
  def get_user_input(message)
    puts message
    gets.chomp
  end


def create_person  
  person_type = get_user_input('Do you want to create a student (1) or a teacher (2)? [Input a number]:')

  case person_type
  when 1
    create_student
  when 2
    create_teacher
  else
    invalid_input
  end
end
```